inc/switchtec/mfg.h

```diff
@@ -89,6 +89,7 @@ struct switchtec_security_cfg_state {
 	uint32_t public_key_exponent;
 	uint32_t public_key_num;
 	uint32_t public_key_ver;
+	uint32_t spi_core_clk_high;
```

Comment:
Why not expand `switchtec_spi_clk_rate` in the structure? I don't see why this new field is necessary.

Reply:
Yes, I agree.

Reply:
`enum switchtec_spi_clk_rate spi_clk_rate` can take value from [1-10]. By default, these values map to [100 - 18.18MB] clock rate. 

On GEN5, there is a bootstrap pin (or consider it as a _turbo_ key), when set, the same value [1-10] maps to [120 - 21.8MB] clock rate. The bootstrap pin state is returned in `spi_core_clk_high` during MRPC. 

So from the chip's perspective, it's the same set of values [1-10] that are interpreted differently, according to the bootstrap pin, or `spi_core_clk_high` flag. 

I think it's safer if we align with this view, otherwise it adds overhead (translation in `cfg_read` function, translation-back in `config_set` function), and could cause more pain in the future.

The `config-set` function will be submitted later.

Reply:
I'd make a change to store the SPI rate as a float inside switchtec_security_cfg_set, so these details can be managed in the library -- instead of having all these hardware specific details leaking into the CLI where they don't belong and just complicate the print functions.

The CLI shouldn't contain a mapping between a hardware specific number and the actual useful piece of information.

Upon revisiting, I find a lot of the mfg structs confusing. They can't be hardware specifc structs because there are enums in some of them, but then why are they using fixed size types (uint32, etc)?